### PR TITLE
Run new application in Docker

### LINKED ISSUE #2936: MapfishPrint 3.8 - class override - ThreadPoolJobManager has been moved to new impl subfolder
when using MapfishPrint 3.8
in https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/scaffolds/create/print/WEB-INF/classes/mapfish-spring-application-context-override.xml 
the path to ThreadPoolJobManager is now 
org.mapfish.print.servlet.job.impl.
instead of
org.mapfish.print.servlet.job.

### LINKED ISSUE #3061: Use environement variable to do the component connexions
From WSGI and mapserver

### LINKED ISSUE #3063: Enable gunicorn access log
` --access-logfile -`?

### LINKED ISSUE #3065: Reactivate phontom js routes check

### LINKED ISSUE #3066: Configure the runtime environment variable for OpenShift

### LINKED ISSUE #3104: Upgrade mapfish-print version in master
Master is currently using an "old" version of mapfish-print. Master should be updated to the most recent version of print, but only after:

- the docker branch has been integrated into master, see https://github.com/camptocamp/c2cgeoportal/pull/3070
- the newest planned version of mapfish-print has been released, see https://github.com/mapfish/mapfish-print/pull/594

Necessary steps on upgrading mapfish-print in c2cgeoportal up to version 2.2 are documented here: https://confluence.camptocamp.com/confluence/display/GS/Update+MapFish+Print+to+3.10.2

### PR DESCRIPTION
**Pull request by sbrunner**
Use Python 3.6
Move tests out of c2cgeoportal
Removes Apache and Deploy configuration

fix #2936
fix #3063
fix #3066
fix #3065
fix #3061
fix #3104

### CODE REVIEW

**Comment by sbrunner** [comment]
The errors from codacy are for me not important (for one), all the others will already exists...

File: c2cgeoportal/scaffolds/create/development.ini.mako_tmpl [hunk 1]
```diff
@@ -14,31 +14,12 @@ authtkt_cookie_name = ${authtkt["cookie_name"]}
 % if "timeout" in authtkt:
 authtkt_timeout = ${authtkt["timeout"]}
 % endif
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/create/development.ini.mako_tmpl]: Do we still use the development.ini file in docker?

File: c2cgeoportal/scaffolds/create/docker-run_tmpl [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "SRID",
+        "EXTENT",
+    ]:
+        docker_cmd.append("--env={}={}".format(env, os.environ.get(env, "")))
+
+    if root:
+        docker_cmd.extend(["--entrypoint", args[0]])
+        args = args[1:]
+
+    docker_cmd.append("camptocamp/geomapfish_build:{{geomapfish_version}}")
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/create/docker-run_tmpl]: Shouldn't this script be in update?

File: c2cgeoportal/scaffolds/create/gunicorn-run_tmpl [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/create/gunicorn-run_tmpl]: Why a template?

File: c2cgeoportal/scaffolds/create/print/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FROM camptocamp/mapfish_print:3.10.0
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/create/print/Dockerfile]: 3.10.2, please

File: c2cgeoportal/views/mapserverproxy.py [hunk 5]
```diff
@@ -102,17 +102,15 @@ def proxy(self):
...
 
                 # In some application we want to display the features owned by a user
                 # than we need his id.
-                if not self.external:
-                    self.params["user_id"] = self.user.id  # pragma: no cover
+                self.params["user_id"] = self.user.id  # pragma: no cover
             else:  # pragma nocover
-                log.warning("The user '{}' has no {}role".format(
-                    self.user.name, "external " if self.external else ""))
+                log.warning("The user '{}' has no role".format(self.user.name))
```
Comment (pvalsecc) [inline c2cgeoportal/views/mapserverproxy.py]: log.warning("The user '%s' has no role", self.user.name)

File: doc/developer/debugging.rst [hunk 6]
```diff
@@ -25,7 +25,7 @@ If the ``pyramid_debugtoolbar`` is enabled the error is directly returned
 in the query that fails.
 
 For print-related issues have a look at the logs in the
-``/srv/tomcat/tomcat1/logs/<instanceid>.log`` file.
+``/srv/tomcat/tomcat1/logs/*.log`` file.
```
Comment (pvalsecc) [inline doc/developer/debugging.rst]: The print docker image doesn't keep the log files. Everything is in stdout

**Comment by sbrunner** [comment]
the parent schema / external themes is an old concept that will be removed...

File: c2cgeoportal/scaffolds/create/docker-run_tmpl (hunk 2 above)
Comment (sbrunner) [reply c2cgeoportal/scaffolds/create/docker-run_tmpl]: I will change the upgrade procedure ant this will be easy to be upgraded from the create template.

**Comment by sbrunner** [comment]
Everything fixed :-)

File: c2cgeoportal/scaffolds/nondockercreate/nondocker.mk [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+# Apache
+
+APACHE_VHOST ?= $(PACKAGE)
+APACHE_CONF_DIR ?= /var/www/vhosts/$(APACHE_VHOST)/conf
+APACHE_GRACEFUL ?= sudo /usr/sbin/apache2ctl graceful
+
+
+.PHONY: clean-all
+clean-all:
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/nondockercreate/nondocker.mk]: You really want that as default target?

File: c2cgeoportal/scaffolds/nondockercreate/vars_nondocker.yaml_tmpl [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        print_url: http://localhost:8080/print-{instanceid}/print/{package}
+    servers:
+        internal: http://localhost/{instanceid}/
+        local: '{web_protocol}://{host}/{instanceid}/'
+        mapserver: http://localhost/{instanceid}/mapserv
+    tiles_url:
+    - '{web_protocol}:://a.tiles.{host}/{instanceid}/ttiles'
+    - '{web_protocol}:://b.tiles.{host}/{instanceid}/ttiles'
+    - '{web_protocol}:://c.tiles.{host}/{instanceid}/ttiles'
+    - '{web_protocol}:://d.tiles.{host}/{instanceid}/ttiles'
```
Comment (pvalsecc) [inline c2cgeoportal/scaffolds/nondockercreate/vars_nondocker.yaml_tmpl]: ttiles -> tiles

File: c2cgeoportal/scaffolds/nondockercreate/nondocker.mk (hunk 7 above)
Comment (sbrunner) [reply c2cgeoportal/scaffolds/nondockercreate/nondocker.mk]: fixed :-)

File: c2cgeoportal/scaffolds/nondockercreate/vars_nondocker.yaml_tmpl (hunk 8 above)
Comment (sbrunner) [reply c2cgeoportal/scaffolds/nondockercreate/vars_nondocker.yaml_tmpl]: Done :-)

**Comment by pvalsecc** [comment]
I cannot make that work. Here is what I'm trying to do:
```
$ travis/docker-get --tag camptocamp/geomapfish_build_dev:latest --folder docker/build --paths docker/build
$ travis/docker-get --tag camptocamp/geomapfish_build:latest --folder . --rules transifex-get buildlocales c2cgeoportal/scaffolds/update/CONST_create_template/ c2cgeoportal/scaffolds/nondockerupdate/CONST_create_template/ --paths docker/build setup.py README.rst c2cgeoportal/ docker/build/fixversions.txt
$ ./docker-run make build
c2c-template --cache /build/c2ctemplate-cache.json --engine mako --files .build/venv/lib/python2.7/site-packages/fa/jquery/templates/index.mako
Traceback (most recent call last):
  File "/usr/local/bin/c2c-template", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/c2c/template.py", line 242, in main
    _proceed(files, used_vars, options)
  File "/usr/local/lib/python3.6/site-packages/c2c/template.py", line 266, in _proceed
    bottle_template(files, used_vars, engine)
  File "/usr/local/lib/python3.6/site-packages/c2c/template.py", line 298, in bottle_template
    template, **used_vars
  File "/usr/local/bin/bottle.py", line 3619, in template
    return TEMPLATES[tplid].render(kwargs)
  File "/usr/local/bin/bottle.py", line 3290, in render
    return self.tpl.render(**_defaults)
  File "/usr/local/lib/python3.6/site-packages/mako/template.py", line 462, in render
    return runtime._render(self, self.callable_, args, data)
  File "/usr/local/lib/python3.6/site-packages/mako/runtime.py", line 838, in _render
    **_kwargs_for_callable(callable_, data))
  File "/usr/local/lib/python3.6/site-packages/mako/runtime.py", line 873, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/usr/local/lib/python3.6/site-packages/mako/runtime.py", line 899, in _exec_template
    callable_(context, *args, **kwargs)
  File "_build_venv_lib_python2_7_site_packages_fa_jquery_templates_index_mako", line 28, in render_body
AttributeError: 'Undefined' object has no attribute 'inline_flash'
make: *** [.build/venv/lib/python2.7/site-packages/fa/jquery/templates/index] Error 1
Makefile:266: recipe for target '.build/venv/lib/python2.7/site-packages/fa/jquery/templates/index' failed
```

**Pull request merged**